windows.js

```diff
@@ -58,36 +62,37 @@ function latestHostedRuby() {
   return tc.find('Ruby', latestVersion)
 }
 
-async function setupMingw(hostedRuby, version) {
+async function setupMingw(version) {
   if (version.startsWith('2.2') || version.startsWith('2.3')) {
-    core.exportVariable('SSL_CERT_FILE', `${hostedRuby}\\ssl\\cert.pem`)
+    core.exportVariable('SSL_CERT_FILE', certFile)
   }
 
-  // Link to embedded MSYS2 in hostedRuby
-  const msys2 = 'C:\\msys64'
+  // Remove when Actions Windows image contains MSYS2 install
   if (!fs.existsSync(msys2)) {
-    const hostedMSYS2 = `${hostedRuby}\\msys64`
+    const hostedMSYS2 = `${latestHostedRuby()}\\msys64`
     await exec.exec(`cmd /c mklink /D ${msys2} ${hostedMSYS2}`)
   }
 
   return [`${msys2}\\mingw64\\bin`, `${msys2}\\usr\\bin`]
 }
 
-async function setupMSWin(hostedRuby) {
+async function setupMSWin() {
   // All standard MSVC OpenSSL builds use C:\Program Files\Common Files\SSL
   const certsDir = 'C:\\Program Files\\Common Files\\SSL\\certs'
   if (!fs.existsSync(certsDir)) {
     fs.mkdirSync(certsDir)
   }
 
-  // Copy cert.pem from hosted Ruby
+  // cert.pem location is hard-coded by OpenSSL msvc builds
   const cert = 'C:\\Program Files\\Common Files\\SSL\\cert.pem'
   if (!fs.existsSync(cert)) {
-    const hostedCert = `${hostedRuby}\\ssl\\cert.pem`
-    fs.copyFileSync(hostedCert, cert)
+    fs.copyFileSync(certFile, cert)
   }
-
-  return addVCVARSEnv()
+  let pathAry = addVCVARSEnv()
+  // add MSYS2 paths for misc gnu utilities like bison and ragel
+  pathAry.push(`${msys2}\\mingw64\\bin`)
+  pathAry.push(`${msys2}\\usr\\bin`)
+  return pathAry
```

Comment:
Could you use an array literal with `...` to splat addVCVARSEnv()?
That would be closer to the code above in `setupMingw` and avoid mutating the array in place.

Reply:
I noticed that push allows multiple arguments, so I changed to that?   VCVARSEnv is defined by the bat file, so we have to 'add' two arrays, but this is just adding two items...

Reply:
I meant https://github.com/ruby/setup-ruby/commit/ae39a8d7d26f4a326f81694e5e99ed009d312fdf actually